cmd/kusk/cmd/deploy.go

```diff
@@ -206,9 +207,24 @@ var deployCmd = &cobra.Command{
 func getParsedAndValidatedOpenAPISpec(apiSpecPath string) (string, error) {
 	const KuskExtensionKey = "x-kusk"
 
-	parsedApiSpec, err := spec.NewParser(&openapi3.Loader{IsExternalRefsAllowed: true}).Parse(apiSpecPath)
-	if err != nil {
-		return "", err
+	parsedApiSpec := &openapi3.T{}
+
+	overlay, err := overlays.NewOverlay(apiSpecPath)
+	if err == nil {
+		overlayPath, err := overlay.Apply()
+		if err != nil {
+			return "", err
```

Comment:
Might be a bit useful to return some error message here along the default such as fmt.Errorf("spec.NewParser: path=%v, %w", path, err).